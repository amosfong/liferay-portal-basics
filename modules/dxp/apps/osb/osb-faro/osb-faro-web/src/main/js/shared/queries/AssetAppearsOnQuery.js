import {gql} from 'apollo-boost';

export default gql`
	query AssetAppearsOnQuery(
		$assetId: String!
		$assetType: AssetType!
		$channelId: String
		$rangeEnd: String
		$rangeKey: Int
		$rangeStart: String
		$selectedMetrics: [String]
		$size: Int!
		$start: Int!
		$title: String
	) {
		assetPages(
			assetId: $assetId
			assetType: $assetType
			channelId: $channelId
			rangeEnd: $rangeEnd
			rangeKey: $rangeKey
			rangeStart: $rangeStart
			selectedMetrics: $selectedMetrics
			size: $size
			start: $start
			title: $title
		) {
			assetMetrics {
				assetTitle
				assetId
				selectedMetrics {
					name
					value
				}
			}
			total
		}
	}
`;
