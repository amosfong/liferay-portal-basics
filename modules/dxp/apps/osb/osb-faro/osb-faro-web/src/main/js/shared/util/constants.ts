import {POSITIONS as ALIGN_POSITIONS} from './align';

export enum AcquisitionTypes {
	Channel = 'CHANNEL',
	Referrer = 'REFERRER',
	SourceMedium = 'SOURCE_MEDIUM'
}

export enum ActivityActions {
	Comments = 3,
	Downloads = 0,
	Previews = 4,
	Submissions = 1,
	Visits = 2
}

export enum Applications {
	Contacts = 'contacts',
	Main = 'main'
}

export enum AssetNames {
	BlogViewed = 'blogViewed',
	CommentPosted = 'commentPosted',
	DocumentDownloaded = 'documentDownloaded',
	DocumentPreviewed = 'documentPreviewed',
	FormSubmitted = 'formSubmitted',
	FormViewed = 'formViewed',
	PageViewed = 'pageViewed',
	WebContentViewed = 'webContentViewed'
}

export enum AssetTypes {
	Asset = 'Asset',
	Blog = 'Blog',
	Document = 'Document',
	Form = 'Form',
	Journal = 'Journal',
	WebContent = 'WebContent',
	WebPage = 'Page'
}

export enum ChannelPermissionTypes {
	AllUsers = 0,
	SelectUsers = 1
}

export enum CompositionTypes {
	AccountInterests = 'accountInterests',
	Acquisitions = 'acquisitions',
	IndividualInterests = 'individualInterests',
	SearchTerms = 'searchTerms',
	SegmentInterests = 'individualSegmentInterests',
	SiteInterests = 'siteInterests'
}

export enum ConjunctionKey {
	And = 'and',
	Or = 'or'
}

export enum CredentialTypes {
	OAuth1 = 'OAuth 1 Authentication',
	OAuth2 = 'OAuth 2 Authentication',
	Token = 'Token Authentication'
}

export enum DataSourceDisplayStatuses {
	Active = 'ACTIVE',
	Configuring = 'CONFIGURING',
	DeleteError = 'DELETE_ERROR',
	InDeletion = 'IN_DELETION',
	Inactive = 'INACTIVE'
}

export enum DataSourceProgressStatuses {
	Completed = 'COMPLETED',
	Failed = 'FAILED',
	InProgress = 'IN_PROGRESS',
	Started = 'STARTED'
}

export enum DataSourceStates {
	ActionNeeded = 'ACTION_NEEDED',
	AnalyticsClientConfigurationFailure = 'ANALYTICS_CLIENT_CONFIGURATION_FAILURE',
	CredentialsInvalid = 'CREDENTIALS_INVALID',
	CredentialsValid = 'CREDENTIALS_VALID',
	Disconnected = 'DISCONNECTED',
	InProgressDeleting = 'IN_PROGRESS_DELETING',
	LiferayVersionInvalid = 'LIFERAY_VERSION_INVALID',
	UndefinedError = 'UNDEFINED_ERROR',
	Ready = 'READY',
	UrlInvalid = 'URL_INVALID',
	Unconfigured = 'UNCONFIGURED'
}
export enum DataSourceStatuses {
	Active = 'ACTIVE',
	Inactive = 'INACTIVE'
}

export enum DataSourceTypes {
	Csv = 'CSV',
	Liferay = 'LIFERAY',
	Salesforce = 'SALESFORCE'
}

export enum EntityTypes {
	Account = 0,
	AccountsSegment = 3,
	Asset = 5,
	DataSource = 1,
	Individual = 2,
	IndividualsSegment = 4,
	Page = 6
}

export enum ExpirationPeriod {
	In30Days = '2592000',
	In6Months = '15778800',
	In1Year = '31557600',
	Indefinite = '3155760000'
}

export enum FaroEnv {
	Local = 'local',
	Production = 'prd',
	Staging = 'stg'
}

export enum FieldContexts {
	Custom = 'custom',
	Demographics = 'demographics',
	Interests = 'interests',
	Organization = 'organization'
}

export enum FieldOwnerTypes {
	Account = 'account',
	Individual = 'individual',
	Organization = 'organization'
}

export enum FieldTypes {
	Boolean = 'Boolean',
	Date = 'Date',
	Number = 'Number',
	String = 'Text'
}

export enum GDPRRequestStatuses {
	Completed = 'COMPLETED',
	Error = 'ERROR',
	Expired = 'EXPIRED',
	Pending = 'PENDING',
	Running = 'RUNNING'
}

export enum GDPRRequestTypes {
	Access = 'ACCESS',
	Delete = 'DELETE',
	Suppress = 'SUPPRESS',
	Unsuppress = 'UNSUPPRESS'
}

export enum JobRunDataPeriods {
	Last7Days = 'LAST_7_DAYS',
	Last30Days = 'LAST_30_DAYS',
	Last180Days = 'LAST_180_DAYS',
	Last365Days = 'LAST_365_DAYS'
}

export enum JobRunFrequencies {
	Every7Days = 'EVERY_7_DAYS',
	Every14Days = 'EVERY_14_DAYS',
	Every30Days = 'EVERY_30_DAYS',
	Manual = 'MANUAL'
}

export enum JobRunStatuses {
	Completed = 'COMPLETED',
	Failed = 'FAILED',
	Published = 'PUBLISHED',
	Running = 'RUNNING'
}

export enum JobStatuses {
	Failed = 'FAILED',
	Pending = 'PENDING',
	Ready = 'READY',
	Running = 'RUNNING',
	Scheduled = 'SCHEDULED'
}

export enum JobTypes {
	ItemSimilarity = 'CONTENT_RECOMMENDATION_ITEM_SIMILARITY'
}

export enum LanguageIds {
	English = 'en_US',
	Japanese = 'ja_JP',
	Portuguese = 'pt_BR',
	Spanish = 'es_ES'
}

export enum OrderByDirections {
	Ascending = 'ASC',
	Descending = 'DESC'
}

export enum PreferencesScopes {
	User = 'user',
	Group = 'group'
}

export enum ProjectStates {
	Activating = 'ACTIVATING',
	AutoRedeployFailed = 'AUTO_REDEPLOY_FAILED',
	Deactivated = 'DEACTIVATED',
	Maintenance = 'MAINTENANCE',
	NotReady = 'NOT READY',
	Ready = 'READY',
	Scheduled = 'SCHEDULED',
	Unavailable = 'UNAVAILABLE',
	Unconfigured = 'UNCONFIGURED'
}

export enum RangeKeyTimeRanges {
	CustomRange = 'CUSTOM',
	Last180Days = '180',
	Last24Hours = '0',
	Last28Days = '28',
	Last30Days = '30',
	Last7Days = '7',
	Last90Days = '90',
	LastYear = '365',
	Yesterday = '1'
}

export enum SegmentStates {
	Disabled = 'DISABLED',
	InProgress = 'IN_PROGRESS',
	Ready = 'READY'
}

export enum SegmentTypes {
	Dynamic = 'DYNAMIC',
	Static = 'STATIC'
}

export enum SessionEntityTypes {
	Account = 'ACCOUNT',
	Individual = 'INDIVIDUAL'
}

export enum Sizes {
	Small = 'sm',
	Medium = 'md',
	Large = 'lg',
	XLarge = 'xl',
	XXLarge = 'xxl',
	XXXLarge = 'xxxl'
}

export enum SubscriptionStatuses {
	Approaching = 1,
	Ok = 0,
	Over = 2
}

export enum TimeIntervals {
	Day = 'day',
	Month = 'month',
	Quarter = 'quarter',
	Week = 'week',
	Year = 'year'
}

export enum TimeSpans {
	AllTime = 'ever',
	LastYear = 'lastYear',
	Last24Hours = 'last24Hours',
	Last7Days = 'last7Days',
	Last28Days = 'last28Days',
	Last30Days = 'last30Days',
	Last90Days = 'last90Days',
	Today = 'today',
	Yesterday = 'yesterday'
}

export enum UserRoleNames {
	Administrator = 'Site Administrator',
	Member = 'Site Member',
	Owner = 'Site Owner'
}

export enum UserStatuses {
	Approved = 0,
	Pending = 1,
	Requested = 2
}

export const ALIGNMENTS_MAP = {
	bottom: ALIGN_POSITIONS.BottomCenter,
	'bottom-left': ALIGN_POSITIONS.BottomLeft,
	'bottom-right': ALIGN_POSITIONS.BottomRight,
	left: ALIGN_POSITIONS.LeftCenter,
	right: ALIGN_POSITIONS.RightCenter,
	top: ALIGN_POSITIONS.TopCenter,
	'top-left': ALIGN_POSITIONS.TopLeft,
	'top-right': ALIGN_POSITIONS.TopRight
};

export const ASSET_METRICS = [
	{
		key: 'abandonmentsMetric',
		selectTitle: `${Liferay.Language.get(
			'form-abandonment'
		)} (${Liferay.Language.get('ratio').toLowerCase()})`,
		title: Liferay.Language.get('form-abandonment'),
		type: 'percentage'
	},
	{
		key: 'clicksMetric',
		selectTitle: `${Liferay.Language.get(
			'asset-clicks'
		)} (${Liferay.Language.get('sum').toLowerCase()})`,
		title: Liferay.Language.get('clicks'),
		type: 'number'
	},
	{
		key: 'completionTimeMetric',
		selectTitle: `${Liferay.Language.get(
			'form-completion-time'
		)} (${Liferay.Language.get('average').toLowerCase()})`,
		title: Liferay.Language.get('completion-time'),
		type: 'time'
	},
	{
		key: 'downloadsMetric',
		selectTitle: `${Liferay.Language.get(
			'asset-downloads'
		)} (${Liferay.Language.get('sum').toLowerCase()})`,
		title: Liferay.Language.get('downloads'),
		type: 'number'
	},
	{
		key: 'readingTimeMetric',
		selectTitle: `${Liferay.Language.get(
			'asset-interaction-time'
		)} (${Liferay.Language.get('average').toLowerCase()})`,
		title: Liferay.Language.get('interaction-time'),
		type: 'time'
	},
	{
		key: 'submissionsMetric',
		selectTitle: `${Liferay.Language.get(
			'form-submissions'
		)} (${Liferay.Language.get('sum').toLowerCase()})`,
		title: Liferay.Language.get('form-submissions'),
		type: 'number'
	},
	{
		key: 'viewsMetric',
		selectTitle: `${Liferay.Language.get(
			'asset-views'
		)} (${Liferay.Language.get('sum').toLowerCase()})`,
		title: Liferay.Language.get('views'),
		type: 'number'
	}
];

export const DATA_RETENTION_PERIOD_KEY = 'data-retention-period';

export const DEVELOPER_MODE = FARO_DEV_MODE;

// LRAC-11571 Disable temporarily Accounts

export const ENABLE_ACCOUNTS = false;

export const ENABLE_ADD_TRIAL_WORKSPACE =
	FARO_ENV === FaroEnv.Local || FARO_ENV === FaroEnv.Staging;

// LRAC - 11652 Hide Displayed Asset Card on the Page report

export const ENABLE_ASSET_CARD = false;

// LRAC-13649 Hide the keywords blocklist screen from Settings>Definitions

export const ENABLE_BLOCKLIST_KEYWORDS = false;

// LRAC-11651 Disable temporarily CSV File

export const ENABLE_CSVFILE = false;

// LRAC-13389 Disable temporarily Delete Property and Delete Data Source buttons

export const ENABLE_DELETE_DATA_SOURCE_BUTTON = false;

export const ENABLE_DELETE_PROPERTY_BUTTON = false;

// LRAC-11650 Hide Form Abandonment Card

export const ENABLE_FORM_ABANDONMENT = false;

// LRAC-10908 Hide global filters

export const ENABLE_GLOBAL_FILTER = false;

// LRAC-13781 [BUG] The Last Seen column only displays the creation date of the token

export const ENABLE_LAST_ACCESS_DATE = false;

// LRAC-11571 Disable temporarily Salesforce

export const ENABLE_SALESFORCE = false;

export const EXPIRATION_DATE_LABELS = {
	[ExpirationPeriod.In30Days]: Liferay.Language.get('30-days'),
	[ExpirationPeriod.In6Months]: Liferay.Language.get('6-months'),
	[ExpirationPeriod.In1Year]: Liferay.Language.get('1-year'),
	[ExpirationPeriod.Indefinite]: Liferay.Language.get('indefinite')
};

export const LANGUAGES = [
	{
		id: LanguageIds.English,
		label: Liferay.Language.get('english')
	},
	{
		id: LanguageIds.Japanese,
		label: Liferay.Language.get('japanese')
	},
	{
		id: LanguageIds.Portuguese,
		label: Liferay.Language.get('portuguese')
	},
	{
		id: LanguageIds.Spanish,
		label: Liferay.Language.get('spanish')
	}
];

export const ONE_DAY = '86400000';

export const ONE_MONTH = '2592000000';

export const POSITIONS = [
	'top',
	'top',
	'right',
	'bottom',
	'bottom',
	'bottom',
	'left',
	'top'
];

export const SEVEN_MONTHS = '18144000000';

export const THIRTEEN_MONTHS = '33696000000';

export const TIME_RANGE_LABELS = {
	[RangeKeyTimeRanges.Last180Days]: Liferay.Language.get('last-180-days'),
	[RangeKeyTimeRanges.Last24Hours]: Liferay.Language.get('last-24-hours'),
	[RangeKeyTimeRanges.Last28Days]: Liferay.Language.get('last-28-days'),
	[RangeKeyTimeRanges.Last30Days]: Liferay.Language.get('last-30-days'),
	[RangeKeyTimeRanges.Last7Days]: Liferay.Language.get('last-seven-days'),
	[RangeKeyTimeRanges.Last90Days]: Liferay.Language.get('last-90-days'),
	[RangeKeyTimeRanges.LastYear]: Liferay.Language.get('last-year'),
	[RangeKeyTimeRanges.Yesterday]: Liferay.Language.get('yesterday')
};

export const TWO_DAYS = '172800000';

const Constants: Window['faroConstants'] = window.faroConstants;

export default Constants;
