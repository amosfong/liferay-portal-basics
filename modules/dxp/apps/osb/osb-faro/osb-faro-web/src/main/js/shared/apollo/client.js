import cache from './cache';
import {ApolloClient} from 'apollo-client';
import {DEVELOPER_MODE} from 'shared/util/constants';
import {get} from 'lodash';
import {HttpLink} from 'apollo-link-http';
import {onError} from 'apollo-link-error';
import {reloadPage} from 'shared/util/router';
import {resolvers} from './resolvers/resolvers';

const groupIdRegex = /^\/workspace\/([a-z0-9._-]+)/;

/**
 * Fecth With Project ID
 * @param {string} uri
 * @param {object} options
 */
const fetchWithGroupId = (uri, options) => {
	const currentUri = new URL(window.location.href);
	const {operationName} = JSON.parse(options.body);
	const matches = currentUri.pathname.match(groupIdRegex);

	if (matches !== null && matches.length > 1) {
		const groupId = matches[1];

		return fetch(
			`${uri}?opname=${operationName}&projectGroupId=${groupId}`,
			options
		);
	}

	return fetch(`${uri}/&opname=${operationName}`, options);
};

const client = new ApolloClient({
	addTypename: true,
	cache,
	connectToDevTools: FARO_DEV_MODE,
	defaultOptions: {
		watchQuery: {
			notifyOnNetworkStatusChange: true
		}
	},
	link: HttpLink.from([
		onError(({operation}) => {
			const status = get(operation.getContext(), ['response', 'status']);

			if (status === 401) {
				reloadPage();
			}
		}),
		new HttpLink({
			credentials: 'same-origin',
			fetch: fetchWithGroupId,
			uri: '/o/cerebro/graphql'
		})
	]),

	resolvers: {
		/**
		 * Queries must render only in the dev mode to avoid
		 * add unnecessary code in the final bundle
		 */
		Query: DEVELOPER_MODE ? resolvers : {}
	}
});

export default client;
