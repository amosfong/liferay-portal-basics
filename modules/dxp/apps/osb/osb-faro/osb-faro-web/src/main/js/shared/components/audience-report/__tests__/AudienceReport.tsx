import AudienceReport from '../AudienceReport';
import client from 'shared/apollo/client';
import React from 'react';
import {ApolloProvider} from '@apollo/react-hooks';
import {cleanup, fireEvent, render} from '@testing-library/react';
import {MetricName} from 'shared/types/MetricName';
import {
	mockAudienceReportReq,
	mockPreferenceReq,
	mockTimeRangeReq
} from 'test/graphql-data';
import {MockedProvider} from '@apollo/react-testing';
import {Name} from '../types';
import {PageAudienceReportQuery} from '../queries';
import {RangeKeyTimeRanges} from 'shared/util/constants';
import {StaticRouter} from 'react-router-dom';
import {waitForLoadingToBeRemoved} from 'test/helpers';

jest.unmock('react-dom');

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useParams: () => ({
		channelId: '456',
		query: {
			rangeKey: RangeKeyTimeRanges.Last30Days
		},
		title: 'Home Page',
		touchpoint: 'https://www.liferay.com'
	})
}));

/**
 * Override Recharts Responsive Container
 * width dimensions fixed to be able to render charts
 */

const tooltipEnabled = jest.fn();

jest.mock('recharts', () => {
	const OriginalModule = jest.requireActual('recharts');

	return {
		...OriginalModule,
		ResponsiveContainer: ({children}) => (
			<OriginalModule.ResponsiveContainer height={350} width={800}>
				{children}
			</OriginalModule.ResponsiveContainer>
		),
		Tooltip: ({children, ...props}) => {
			if (props.active) {
				tooltipEnabled();
			}

			return (
				<OriginalModule.Tooltip {...props} active>
					{children}
				</OriginalModule.Tooltip>
			);
		}
	};
});

const WrappedComponent = ({queryProps}) => (
	<ApolloProvider client={client}>
		<StaticRouter>
			<MockedProvider
				mocks={[
					mockTimeRangeReq(),
					mockPreferenceReq(),
					mockAudienceReportReq({queryProps})
				]}
			>
				<AudienceReport
					filters={{devices: [], location: []}}
					mapper={result =>
						result?.[queryProps.name]?.[queryProps.metricName]
					}
					name={Name.Page}
					Query={PageAudienceReportQuery(queryProps)}
					rangeSelectors={{
						rangeEnd: '',
						rangeKey: RangeKeyTimeRanges.Last30Days,
						rangeStart: ''
					}}
				/>
			</MockedProvider>
		</StaticRouter>
	</ApolloProvider>
);

describe('CommerceMetricCard', () => {
	afterEach(cleanup);

	it('should render', async () => {
		const {container} = render(
			<WrappedComponent
				queryProps={{
					metricName: MetricName.Views,
					name: Name.Page
				}}
			/>
		);

		await waitForLoadingToBeRemoved(container);

		expect(container).toMatchSnapshot();
	});

	it('should render a tooltip when donuts mouse over', async () => {
		const {container} = render(
			<WrappedComponent
				queryProps={{
					metricName: MetricName.Views,
					name: Name.Page
				}}
			/>
		);

		await waitForLoadingToBeRemoved(container);

		const donut = document.querySelector('.recharts-pie-sector');

		expect(donut).toBeInTheDocument();

		fireEvent.mouseEnter(donut);

		expect(tooltipEnabled).toHaveBeenCalledTimes(1);
	});
});
