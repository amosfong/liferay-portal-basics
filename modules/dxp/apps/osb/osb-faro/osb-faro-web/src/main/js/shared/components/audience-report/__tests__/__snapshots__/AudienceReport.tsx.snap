// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommerceMetricCard should render 1`] = `
<div>
  <div
    class="audience-report-chart row w-100"
  >
    <div
      class="col-sm-6"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-6 text-center"
        >
          <div
            class="d-inline-flex gap"
          >
            <div
              class="h4 mb-3 mr-2 text-center text-secondary title"
            >
              Visitors
            </div>
          </div>
          <div
            class="audience-report-chart-donut"
          >
            <div
              class="recharts-responsive-container"
              style="width: 800px; height: 350px; min-width: 0;"
            >
              <div
                class="recharts-wrapper"
                role="region"
                style="position: relative; cursor: default; width: 800px; height: 350px;"
              >
                <svg
                  class="recharts-surface"
                  cx="50%"
                  cy="50%"
                  height="350"
                  viewBox="0 0 800 350"
                  width="800"
                >
                  <title />
                  <desc />
                  <defs>
                    <clippath
                      id="recharts1-clip"
                    >
                      <rect
                        height="340"
                        width="790"
                        x="5"
                        y="5"
                      />
                    </clippath>
                  </defs>
                  <g
                    class="recharts-layer recharts-pie"
                    tabindex="0"
                  >
                    <g
                      class="recharts-layer recharts-pie-sector"
                      tabindex="-1"
                    >
                      <path
                        class="recharts-sector"
                        color="#FFB46E"
                        cx="400"
                        cy="147"
                        d="M 400,-6
    A 153,153,0,
    1,1,
    392.70079671790785,-5.825788502617229
  L 395.94488706550436,62.09678416521264
            A 85,85,0,
            1,0,
            400,62 Z"
                        fill="#FFB46E"
                        fill-opacity="1"
                        name="0"
                        role="img"
                        stroke="#FFB46E"
                        stroke-opacity="1"
                        tabindex="-1"
                      />
                    </g>
                    <g
                      class="recharts-layer recharts-pie-sector"
                      tabindex="-1"
                    >
                      <path
                        class="recharts-sector"
                        color="#4B9BFF"
                        cx="400"
                        cy="147"
                        d="M 392.70079671790785,-5.825788502617229
    A 153,153,0,
    0,1,
    400,-6
  L 400,62
            A 85,85,0,
            0,0,
            395.94488706550436,62.09678416521264 Z"
                        fill="#4B9BFF"
                        fill-opacity="1"
                        name="1"
                        role="img"
                        stroke="#4B9BFF"
                        stroke-opacity="1"
                        tabindex="-1"
                      />
                    </g>
                    <g
                      class="recharts-layer recharts-pie-labels"
                    >
                      <g
                        class="recharts-layer"
                      >
                        <text
                          class="recharts-text"
                          fill="#808080"
                          style="fill: black; font: Symbol\\" 14px 14px; font-size: 1rem; font-weight: 600;"
                          text-anchor="middle"
                          x="402.8393874272831"
                          y="265.96612071946277"
                        >
                          <tspan
                            dy="0em"
                            x="402.8393874272831"
                          >
                            99.24%
                          </tspan>
                        </text>
                      </g>
                      <g
                        class="recharts-layer"
                      >
                        <text
                          class="recharts-text"
                          fill="#808080"
                          style="fill: black; font: Symbol\\" 14px 14px; font-size: 1rem; font-weight: 600;"
                          text-anchor="middle"
                          x="397.16061257271696"
                          y="28.033879280537235"
                        >
                          <tspan
                            dy="0em"
                            x="397.16061257271696"
                          >
                            0.76%
                          </tspan>
                        </text>
                      </g>
                    </g>
                    <text
                      class="recharts-text recharts-label"
                      fill="#808080"
                      offset="5"
                      text-anchor="middle"
                      x="400"
                      y="147"
                    >
                      <tspan
                        dy="0.355em"
                        x="400"
                      >
                        12,902
                      </tspan>
                    </text>
                  </g>
                </svg>
                <div
                  class="recharts-legend-wrapper"
                  style="position: absolute; width: auto; height: auto; left: 400px; bottom: 5px;"
                >
                  <ul
                    class="recharts-default-legend"
                    style="padding: 0px; margin: 0px; text-align: left;"
                  >
                    <li
                      class="recharts-legend-item legend-item-0"
                      style="display: block; margin-right: 10px;"
                    >
                      <svg
                        class="recharts-surface"
                        height="12"
                        style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                        viewBox="0 0 32 32"
                        width="12"
                      >
                        <title />
                        <desc />
                        <path
                          class="recharts-symbols"
                          cx="16"
                          cy="16"
                          d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                          fill="#FFB46E"
                          transform="translate(16, 16)"
                        />
                      </svg>
                      <span
                        class="recharts-legend-item-text"
                        style="color: rgb(255, 180, 110);"
                      >
                        <span
                          class="text-3 text-secondary"
                        >
                          Anonymous Individuals
                        </span>
                      </span>
                    </li>
                    <li
                      class="recharts-legend-item legend-item-1"
                      style="display: block; margin-right: 10px;"
                    >
                      <svg
                        class="recharts-surface"
                        height="12"
                        style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                        viewBox="0 0 32 32"
                        width="12"
                      >
                        <title />
                        <desc />
                        <path
                          class="recharts-symbols"
                          cx="16"
                          cy="16"
                          d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                          fill="#4B9BFF"
                          transform="translate(16, 16)"
                        />
                      </svg>
                      <span
                        class="recharts-legend-item-text"
                        style="color: rgb(75, 155, 255);"
                      >
                        <span
                          class="text-3 text-secondary"
                        >
                          Known Individuals
                        </span>
                      </span>
                    </li>
                  </ul>
                </div>
                <div
                  class="recharts-tooltip-wrapper"
                  role="dialog"
                  style="transition: transform 400ms ease; transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
                  tabindex="-1"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-sm-6 text-center"
        >
          <div
            class="d-inline-flex gap"
          >
            <div
              class="h4 mb-3 mr-2 text-center text-secondary title"
            />
            <span
              class="info-popover-root"
            >
              <svg
                class="lexicon-icon lexicon-icon-question-circle-full icon-root"
                role="presentation"
              >
                <use
                  href="#question-circle-full"
                />
              </svg>
            </span>
          </div>
          <div
            class="audience-report-chart-donut"
          >
            <div
              class="recharts-responsive-container"
              style="width: 800px; height: 350px; min-width: 0;"
            >
              <div
                class="recharts-wrapper"
                role="region"
                style="position: relative; cursor: default; width: 800px; height: 350px;"
              >
                <svg
                  class="recharts-surface"
                  cx="50%"
                  cy="50%"
                  height="350"
                  viewBox="0 0 800 350"
                  width="800"
                >
                  <title />
                  <desc />
                  <defs>
                    <clippath
                      id="recharts3-clip"
                    >
                      <rect
                        height="340"
                        width="790"
                        x="5"
                        y="5"
                      />
                    </clippath>
                  </defs>
                  <g
                    class="recharts-layer recharts-pie"
                    tabindex="0"
                  >
                    <g
                      class="recharts-layer recharts-pie-sector"
                      tabindex="-1"
                    >
                      <path
                        class="recharts-sector"
                        color="#A1E7CC"
                        cx="400"
                        cy="147"
                        d="M 400,-6
    A 153,153,0,
    0,1,
    419.56520573772946,-4.743872114956645
  L 410.869558743183,62.69784882502408
            A 85,85,0,
            0,0,
            400,62 Z"
                        fill="#A1E7CC"
                        fill-opacity="1"
                        name="0"
                        role="img"
                        stroke="#A1E7CC"
                        stroke-opacity="1"
                        tabindex="-1"
                      />
                    </g>
                    <g
                      class="recharts-layer recharts-pie-sector"
                      tabindex="-1"
                    >
                      <path
                        class="recharts-sector"
                        color="#26966B"
                        cx="400"
                        cy="147"
                        d="M 419.56520573772946,-4.743872114956645
    A 153,153,0,
    1,1,
    400,-6
  L 400,62
            A 85,85,0,
            1,0,
            410.869558743183,62.69784882502408 Z"
                        fill="#26966B"
                        fill-opacity="1"
                        name="1"
                        role="img"
                        stroke="#26966B"
                        stroke-opacity="1"
                        tabindex="-1"
                      />
                    </g>
                    <g
                      class="recharts-layer recharts-pie-labels"
                    >
                      <g
                        class="recharts-layer"
                      >
                        <text
                          class="recharts-text"
                          fill="#808080"
                          style="fill: black; font: Symbol\\" 14px 14px; font-size: 1rem; font-weight: 600;"
                          text-anchor="middle"
                          x="407.62435617770484"
                          y="28.244498262709982"
                        >
                          <tspan
                            dy="0em"
                            x="407.62435617770484"
                          >
                            2.04%
                          </tspan>
                        </text>
                      </g>
                      <g
                        class="recharts-layer"
                      >
                        <text
                          class="recharts-text"
                          fill="#808080"
                          style="fill: black; font: Symbol\\" 14px 14px; font-size: 1rem; font-weight: 600;"
                          text-anchor="middle"
                          x="392.37564382229516"
                          y="265.75550173729005"
                        >
                          <tspan
                            dy="0em"
                            x="392.37564382229516"
                          >
                            97.96%
                          </tspan>
                        </text>
                      </g>
                    </g>
                    <text
                      class="recharts-text recharts-label"
                      fill="#808080"
                      offset="5"
                      text-anchor="middle"
                      x="400"
                      y="147"
                    >
                      <tspan
                        dy="0.355em"
                        x="400"
                      >
                        98
                      </tspan>
                    </text>
                  </g>
                </svg>
                <div
                  class="recharts-legend-wrapper"
                  style="position: absolute; width: auto; height: auto; left: 400px; bottom: 5px;"
                >
                  <ul
                    class="recharts-default-legend"
                    style="padding: 0px; margin: 0px; text-align: left;"
                  >
                    <li
                      class="recharts-legend-item legend-item-0"
                      style="display: block; margin-right: 10px;"
                    >
                      <svg
                        class="recharts-surface"
                        height="12"
                        style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                        viewBox="0 0 32 32"
                        width="12"
                      >
                        <title />
                        <desc />
                        <path
                          class="recharts-symbols"
                          cx="16"
                          cy="16"
                          d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                          fill="#A1E7CC"
                          transform="translate(16, 16)"
                        />
                      </svg>
                      <span
                        class="recharts-legend-item-text"
                        style="color: rgb(161, 231, 204);"
                      >
                        <span
                          class="text-3 text-secondary"
                        >
                          Segmented
                        </span>
                      </span>
                    </li>
                    <li
                      class="recharts-legend-item legend-item-1"
                      style="display: block; margin-right: 10px;"
                    >
                      <svg
                        class="recharts-surface"
                        height="12"
                        style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                        viewBox="0 0 32 32"
                        width="12"
                      >
                        <title />
                        <desc />
                        <path
                          class="recharts-symbols"
                          cx="16"
                          cy="16"
                          d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                          fill="#26966B"
                          transform="translate(16, 16)"
                        />
                      </svg>
                      <span
                        class="recharts-legend-item-text"
                        style="color: rgb(38, 150, 107);"
                      >
                        <span
                          class="text-3 text-secondary"
                        >
                          Unsegmented
                        </span>
                      </span>
                    </li>
                  </ul>
                </div>
                <div
                  class="recharts-tooltip-wrapper"
                  role="dialog"
                  style="transition: transform 400ms ease; transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
                  tabindex="-1"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-sm-6 pl-5"
    >
      <div
        class="d-inline-flex gap"
      >
        <div
          class="h4 mb-3 mr-2 text-center text-secondary title"
        >
          Viewer Segments
        </div>
        <span
          class="info-popover-root"
        >
          <svg
            class="lexicon-icon lexicon-icon-question-circle-full icon-root"
            role="presentation"
          >
            <use
              href="#question-circle-full"
            />
          </svg>
        </span>
      </div>
      <div
        class="audience-report-chart-bar"
      >
        <div
          class="analytics-bar-chart-html disable-scroll"
        >
          <div
            class="analytics-bar-chart-html-grid"
          >
            <div
              class="analytics-bar-chart-html-grid-item"
            >
              <span>
                0%
              </span>
            </div>
            <div
              class="analytics-bar-chart-html-grid-item"
            >
              <span>
                50%
              </span>
            </div>
            <div
              class="analytics-bar-chart-html-grid-item"
            >
              <span>
                100%
              </span>
            </div>
            <div
              class="analytics-bar-chart-html-grid-item"
            >
              <span>
                150%
              </span>
            </div>
          </div>
          <div
            class="analytics-bar-chart-html-group-items"
          >
            <ul
              class="analytics-bar-chart-html-items"
            >
              <li
                class="analytics-bar-chart-html-item"
              >
                <div
                  class="analytics-bar-chart-html-header"
                >
                  <div
                    class="analytics-bar-chart-html-progress"
                  >
                    <div
                      style="background-color: rgb(161, 231, 204); width: 66.7%;"
                    />
                  </div>
                  <div>
                    <div
                      class="analytics-bar-chart-html-column"
                    >
                      <svg
                        class="lexicon-icon lexicon-icon-ac-segment analytics-bar-chart-html-icon icon-root"
                        role="presentation"
                      >
                        <use
                          href="#ac-segment"
                        />
                      </svg>
                      <div
                        class="text-truncate w-100"
                      >
                        <span>
                          UK Visitors
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
